buildSrc/src/main/kotlin/io/spine/internal/gradle/javacompile/ErrorProneConfig.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.javacompile
+
+import io.spine.internal.gradle.javacompile.ErrorProneConfig.configureErrorProne
+import net.ltgt.gradle.errorprone.errorprone
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.kotlin.dsl.withType
+
+/**
+ * Configures `ErrorProne` plugin.
+ *
+ * ```
+ * tasks {
+ *     withType<JavaCompile> {
+ *         configureErrorProne()
+ *     }
+ * }
+ * ```
+ */
+object ErrorProneConfig {
+
+    private val CODE_ANALYZING_FLAGS = listOf(
```

Comment:
`CODE_ANALYZING_FLAGS` — does this name have to be that long? Can it be just `flags` or, better, `args`?

Reply:
Yes, It can. 
But I would like to go with more specific name as long as we add them to the list.
```
.errorproneArgs
.addAll(CODE_ANALYZING_FLAGS)

// => Let's add flags that tune code analysis to the list of arguments.
```
```
.errorproneArgs
.set(ARGUMENTS)

// => Let's set the arguments for ErrorProne.
```

The original script in Groovy was adding them so I have chosen the first option.

Reply:
As long as this is a private constant, and it virtually contains _everything_ that we ever add, we should aim for brevity. Please adjust your naming.